base/compiler/tfuncs.jl

```diff
@@ -3010,10 +3010,6 @@ function abstract_applicable(interp::AbstractInterpreter, argtypes::Vector{Any},
             rt = Const(true) # has applicable matches
         end
         if rt !== Bool
-            for i = 1:napplicable
-                (; match, edges, edge_idx) = applicable[i]
-                edges[edge_idx] = codeinst_as_invoke_edge(interp, specialize_method(match))
-            end
```

Comment:
`codeinst_as_invoke_edge` here was necessary to recode backedges for this method match.

Reply:
I changed the hackmd reference to reflect this change now: no specialization edges (CodeInstance) are required if the only point is to record the existence of the edge (the MethodInstance). The edge computation code was updated to be aware of this point as well, such that it will generate the method match edge as needed, but will then replace that with a CodeInstance edge if that results in redundancies.

Reply:
Yes, I think the `Method`-edge encoding in particular seems useful, where we can distinguish between edges that don’t need to be recorded as backedges when `abstract_call_method` fails and those that do need to be recorded as backedges, such as in `abstract_applicable`.

Reply:
I implemented the `Method`-edge encoding: 3836a39cc1f5c8e2f0939113a2fc0dad4e69f6f8